File: iosgl/iosgl/bt/skill/key/SkillKey.m
//
//  SkillKey.m
//  iosgl
//
//  Created by zhao on 17/4/2020.
//  Copyright © 2020 zhao. All rights reserved.
//

#import "SkillKey.h"
#import "SkillKeyVo.h"
#import "Scene_data.h"
#import "ParticleManager.h"

@implementation SkillKey
-(void)setInfo:(SkillKeyVo*)obj;
{
    SkillKey* this=self;
    this.time=obj.frame*[Scene_data default].frameTime;
    this.particle=  [[ParticleManager default] getParticleByte: obj.url];
 
}
-(void)addToRender;
{
    SkillKey* this=self;
    if (!this.particle){
            return;
        }
    [this.particle reset];
    this.particle.sceneVisible = true;
    [[ParticleManager default] addParticle:this.particle];
}
 

@end

File: android/SceneWorld001/app/src/main/java/z3d/skill/key/SkillKey.java
package z3d.skill.key;

import scene.CallBack;
import z3d.base.Scene_data;
import z3d.display.particle.CombineParticle;
import z3d.filemodel.ParticleManager;
import z3d.skill.vo.SkillKeyVo;

public class SkillKey {
    public float time ;
    public CombineParticle particle;
    public CallBack removeCallFun;

    public void addToRender(ParticleManager particleManager) {
        if (this.particle==null) {
            return;
        }
        this.particle.reset();
        this.particle.sceneVisible = true;
        particleManager.addParticle(this.particle);
    }

    public void setInfo(SkillKeyVo obj) {
        this.time = obj.frame * Scene_data.frameTime;
        this.particle = ParticleManager.getInstance().getParticleByte(Scene_data.fileRoot + obj.url);


    }
}
